docs/product-user-guide/src/main/asciidoc/guided-decision-tables-columns-condition-BRL-con.adoc

```diff
@@ -1,13 +1,19 @@
 [id='guided-decision-tables-columns-condition-BRL-con']
 = "Add a Condition BRL fragment"
 
-A Business Rule Language (BRL) fragment is a section of a rule created using the Guided Rule Editor. The condition BRL fragment is the "WHEN" portion of the rule, and the <<guided-decision-tables-columns-action-BRL-con, Action BRL fragment>> is the "THEN" portion of the rule. With this column option, you can define a Condition BRL fragment to be used in the left ("WHEN") side of a rule. In the embedded Guided Rule Editor, field values defined as "Template Keys" form columns in the decision table.
-Simpler column types can refer to Facts and Fact fields bound in the BRL fragment and vice-versa.
+A Business Rule Language (BRL) fragment is a section of a rule created using the guided rule designer. The condition BRL fragment is the "WHEN" portion of the rule, and the xref:guided-decision-tables-columns-action-BRL-con[action BRL fragment] is the "THEN" portion of the rule. With this column option, you can define a condition BRL fragment to be used in the left ("WHEN") side of a rule. Any *Template key* field values (as opposed to *Literal* or *Formula* values) that you add to conditions (example: `$rate`) form separate columns in the decision table. Simpler column types can refer to Facts and Fact fields bound in the BRL fragment and vice-versa.
 
 Example condition BRL fragment for a loan application:
 
-.Add a Condition BRL fragment
-image::guided-decision-tables-columns-condition-BRL.png[Condition BRL Fragment column for guided decision table editor in BRMS User Guide 6.0.2]
+.Add a condition BRL fragment with the embedded guided rule designer
+image::guided-decision-tables-columns-condition-BRL.png[Condition BRL Fragment column for guided decision table designer]
+
+You can also select *Free form DRL* from the list of condition options to define the condition BRL fragment without the embedded guided rule designer.
```

Comment:
If the User elects to use "Free Form DRL" they can provide template keys (to create columns in the table) by using the ```@{key}``` syntax. For example ```Person( age > @{age} )``` would create a column in the table for "age". It might be worth mentioning this **if** it's not covered elsewhere.

Reply:
It should be noted however that with "Free Form DRL" the column's data-type is **always** Text/String. So, for example ```Person( age > @{age} )``` the User could enter ```abc``` in the cell. Another example is ```Person( name == "@{name}" )``` where the required quotation marks are included in the Free Form DRL.

Reply:
So, I've mentioned Template Keys briefly in the intro paragraph to this very section: file:///home/sterobin/repos/kie-docs/docs/product-assembly_guided-decision-tables/src/main/asciidoc/dm/build/tmp/en-US/html-single/index.html#guided-decision-tables-columns-condition-BRL-con

But a few questions/ambiguities here:
- I'm a bit confused because Template Keys are not specific to free form DRL, as you say. When I use the guided editor and click to add a field, the prompt gives me options for Literal, Template Key, and Formula values. And it's in the format `$key`. Are you just saying that to use Template Keys in the free form DRL, they should not follow that same format, and instead be in the format you've described here? i.e., `@{key} ?

- Also, since we bring this up, can you clarify when the user should use Template Keys? The community doc just says that it serves as a placeholder and creates a new column, as you mention, but so do most other column options. That's the point of adding one or more of these various column types. That is, if I create the same column type with a Literal value instead of a Template Key, it creates a condition column accordingly. I'm not sure therefore when to use or not to use Template Keys.

- And you say that for free form DRL the data type is always String, i.e., text. But what if the data is numerical, like in your example with age? Wouldn't that need to be a data type of Integer?

Reply:
Hi @sterobin 

The explicit Template Key field type (vs Literal, Formula etc) auto-magically creates ```@{$key}``` in the pre-processed DRL before the keys are substituted by values in the final DRL. The prefix $ is to be honest entirely arbitrary.. IDK where it originated from. When using (and only using) "Free Form DRL" should the User enter the form ```@{key}```. Free Form DRL is just plain text and the User can enter whatever they like.. but ```@{key}``` will result in new columns for the values.

The point of using ```@{key}``` template keys is only for Free Form DRL. Since its text can be any value these special place-holders allow for the User to have the text interpolated with real values at rule/DRL generation time. Other constructs create columns too; but they're more visual/guided than plain text. You can enter ```@{key}``` as the literal value in a simple condition column (if the field data-type is ```String```) but it will not be interpolated and will be the real value. For example if you define a simple condition and enter the cell value as ```@{key}``` you'd get DRL like ```Person( name == "@{key}" )```.. the value is literal.

It's a limitation of the workbench that the data-type is always ```String``` and, since use of "Free Form DRL" is an advanced concept (well, catch all for everything the editor does not support in a more visual/guided way) it's left to the User to ensure the text and use of template keys follows the correct syntax. It is only the data-type of the cell in the table; the underlying model data-type can be anything... it's just for the User to know it's, for example, an integer and therefore not to enter alpha-characters.

In an ideal world we'd either (a) determine the real data-type and restrict cell data entry accordingly (like we do for example with simple conditions), (b) provide an expression editor for Free Form DRL that prevents erroneous statements from being entered.. Both come down to (what appears to be) a lack of (time, resources) investment in enhancing the Guided Rule Editor (and hence BRL Fragments) further.. although a replacement has been talked about for years..